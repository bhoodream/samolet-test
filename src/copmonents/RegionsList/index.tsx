import { FC } from 'react';
import { Link } from 'react-router-dom';

import Loading from '../Loading';

const RegionsList: FC<{ items: Region[] }> = ({ items }) => {
    if (!items.length) {
        return <Loading />;
    }

    return (
        <ol>
            {items.map(({ kopuk, fullname }) => (
                <li key={kopuk}>
                    <Link to={`/${kopuk}`}>{fullname}</Link>
                </li>
            ))}
        </ol>
    );
};

export default RegionsList;
