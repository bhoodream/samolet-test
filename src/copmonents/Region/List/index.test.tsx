import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import RegionsList from './index';

it('renders correctly', () => {
    const tree = renderer
        .create(
            <Router>
                <RegionsList
                    onShowMore={() => {}}
                    showItemsCount={10}
                    items={
                        [
                            {
                                subscribers: 123,
                                libraries: 543,
                            },
                            {
                                subscribers: 123,
                                libraries: 543,
                            },
                        ] as Region[]
                    }
                />
            </Router>
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
