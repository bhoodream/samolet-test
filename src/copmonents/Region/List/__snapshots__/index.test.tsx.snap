// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ant-list ant-list-split ant-list-grid"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-row"
        style={
          Object {
            "marginLeft": -8,
            "marginRight": -8,
          }
        }
      >
        <div>
          <div
            className="ant-col"
            style={
              Object {
                "flex": "1 1 auto",
                "paddingLeft": 8,
                "paddingRight": 8,
              }
            }
          >
            <div
              className="ant-list-item"
            >
              <div
                className="ant-card ant-card-bordered"
              >
                <div
                  className="ant-card-head"
                  style={Object {}}
                >
                  <div
                    className="ant-card-head-wrapper"
                  >
                    <div
                      className="ant-card-head-title"
                    >
                      <h4
                        className="ant-typography"
                        style={
                          Object {
                            "WebkitLineClamp": undefined,
                          }
                        }
                      >
                        
                      </h4>
                    </div>
                    <div
                      className="ant-card-extra"
                    >
                      <a
                        href="/undefined"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        <button
                          className="ant-btn"
                          onClick={[Function]}
                          type="button"
                        >
                          <span
                            aria-label="right"
                            className="anticon anticon-right"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="right"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                              />
                            </svg>
                          </span>
                        </button>
                      </a>
                    </div>
                  </div>
                </div>
                <div
                  className="ant-card-body"
                  style={Object {}}
                >
                  <div
                    className="ant-row"
                    style={
                      Object {
                        "marginLeft": -20,
                        "marginRight": -20,
                        "maxWidth": "100%",
                      }
                    }
                  >
                    <div
                      className="ant-col"
                      style={
                        Object {
                          "paddingLeft": 20,
                          "paddingRight": 20,
                        }
                      }
                    >
                      <div
                        className="ant-statistic"
                      >
                        <div
                          className="ant-statistic-title"
                        >
                          <span
                            aria-label="bank"
                            className="anticon anticon-bank"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="bank"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M894 462c30.9 0 43.8-39.7 18.7-58L530.8 126.2a31.81 31.81 0 00-37.6 0L111.3 404c-25.1 18.2-12.2 58 18.8 58H192v374h-72c-4.4 0-8 3.6-8 8v52c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-52c0-4.4-3.6-8-8-8h-72V462h62zM512 196.7l271.1 197.2H240.9L512 196.7zM264 462h117v374H264V462zm189 0h117v374H453V462zm307 374H642V462h118v374z"
                              />
                            </svg>
                          </span>
                           Библиотеки
                        </div>
                        <div
                          className="ant-statistic-content"
                        >
                          <span
                            className="ant-statistic-content-value"
                          >
                            <span
                              className="ant-statistic-content-value-int"
                            >
                              
                              543
                            </span>
                            
                          </span>
                        </div>
                      </div>
                    </div>
                    <div
                      className="ant-col"
                      style={
                        Object {
                          "paddingLeft": 20,
                          "paddingRight": 20,
                        }
                      }
                    >
                      <div
                        className="ant-statistic"
                      >
                        <div
                          className="ant-statistic-title"
                        >
                          <span
                            aria-label="user"
                            className="anticon anticon-user"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="user"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                              />
                            </svg>
                          </span>
                           Читатели
                        </div>
                        <div
                          className="ant-statistic-content"
                        >
                          <span
                            className="ant-statistic-content-value"
                          >
                            <span
                              className="ant-statistic-content-value-int"
                            >
                              
                              123
                            </span>
                            
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            className="ant-col"
            style={
              Object {
                "flex": "1 1 auto",
                "paddingLeft": 8,
                "paddingRight": 8,
              }
            }
          >
            <div
              className="ant-list-item"
            >
              <div
                className="ant-card ant-card-bordered"
              >
                <div
                  className="ant-card-head"
                  style={Object {}}
                >
                  <div
                    className="ant-card-head-wrapper"
                  >
                    <div
                      className="ant-card-head-title"
                    >
                      <h4
                        className="ant-typography"
                        style={
                          Object {
                            "WebkitLineClamp": undefined,
                          }
                        }
                      >
                        
                      </h4>
                    </div>
                    <div
                      className="ant-card-extra"
                    >
                      <a
                        href="/undefined"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        <button
                          className="ant-btn"
                          onClick={[Function]}
                          type="button"
                        >
                          <span
                            aria-label="right"
                            className="anticon anticon-right"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="right"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                              />
                            </svg>
                          </span>
                        </button>
                      </a>
                    </div>
                  </div>
                </div>
                <div
                  className="ant-card-body"
                  style={Object {}}
                >
                  <div
                    className="ant-row"
                    style={
                      Object {
                        "marginLeft": -20,
                        "marginRight": -20,
                        "maxWidth": "100%",
                      }
                    }
                  >
                    <div
                      className="ant-col"
                      style={
                        Object {
                          "paddingLeft": 20,
                          "paddingRight": 20,
                        }
                      }
                    >
                      <div
                        className="ant-statistic"
                      >
                        <div
                          className="ant-statistic-title"
                        >
                          <span
                            aria-label="bank"
                            className="anticon anticon-bank"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="bank"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M894 462c30.9 0 43.8-39.7 18.7-58L530.8 126.2a31.81 31.81 0 00-37.6 0L111.3 404c-25.1 18.2-12.2 58 18.8 58H192v374h-72c-4.4 0-8 3.6-8 8v52c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-52c0-4.4-3.6-8-8-8h-72V462h62zM512 196.7l271.1 197.2H240.9L512 196.7zM264 462h117v374H264V462zm189 0h117v374H453V462zm307 374H642V462h118v374z"
                              />
                            </svg>
                          </span>
                           Библиотеки
                        </div>
                        <div
                          className="ant-statistic-content"
                        >
                          <span
                            className="ant-statistic-content-value"
                          >
                            <span
                              className="ant-statistic-content-value-int"
                            >
                              
                              543
                            </span>
                            
                          </span>
                        </div>
                      </div>
                    </div>
                    <div
                      className="ant-col"
                      style={
                        Object {
                          "paddingLeft": 20,
                          "paddingRight": 20,
                        }
                      }
                    >
                      <div
                        className="ant-statistic"
                      >
                        <div
                          className="ant-statistic-title"
                        >
                          <span
                            aria-label="user"
                            className="anticon anticon-user"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="user"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                              />
                            </svg>
                          </span>
                           Читатели
                        </div>
                        <div
                          className="ant-statistic-content"
                        >
                          <span
                            className="ant-statistic-content-value"
                          >
                            <span
                              className="ant-statistic-content-value-int"
                            >
                              
                              123
                            </span>
                            
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
